tools/build.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--clone_dir", required=True)
+    parser.add_argument("--conda_path")
+    parser.add_argument("--config_mode", default="Shipping")
+    parser.add_argument("--unreal_engine_dir", required=True)
+    parser.add_argument("--num_parallel_jobs", type=int, default=1)
+    parser.add_argument("--target_platform", required=True, help="Values accepted are Win64, Linux, Mac")
+    parser.add_argument("--output_dir", required=True)
+    parser.add_argument("--tag", required=True)
+    args = parser.parse_args()
+
+    assert args.target_platform in ["Linux", "Mac"] and args.conda_path
+
+    # clone repo along with submodules
+    cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", args.clone_dir]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    # change working directory to the cloned directory above
+    cwd = os.getcwd()
+    print(f"[SPEAR | build.py] Changing current working dir from {cwd} to {args.clone_dir}")
+    os.chdir(args.clone_dir)
+
+    # create a new virtual python env called spear-env
+    if sys.platform == "win32":
+        cmd = ["conda", "create", "-n", "spear-env", "python=3.8", "-y"]
+    elif sys.platform in ["linux", "darwin"]:
+        cmd = ["conda create -n spear-env python=3.8 -y"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # install msgpack-rpc-python package in the newly created python env
+    if sys.platform == "win32":
+        cmd = ["conda", "activate", "spear-env&", "pip", "install", "-e", "third_party/msgpack-rpc-python"]
+    elif sys.platform in ["linux", "darwin"]:    
+        cmd = [f". {args.conda_path}; conda activate spear-env; pip install -e third_party/msgpack-rpc-python"]
```

Comment:
It's fine to assume that "conda" is available on the command-line, why do we need conda_path here but not above?

Reply:
When we use `shell=True` on unix systems, it defaults to `/bin/sh`. If your conda setup is not configured to `/bin/sh`'s profile (e.g., on mac, default is `/bin/zsh`, so your conda setup most likely will be configured for `zsh`'s profile), you won't be able to use conda commands. So sourcing conda.sh makes it so that conda can be used on any shell.

The other option would be take as input an argument (args.shell) to the shell's path such as `/bin/sh` or `/bin/zsh` so that we don't need conda_path or the need to source conda.sh as well.

I prefer the second option, what do you think?

Reply:
I thought we agreed to assume that your conda env is already set up.